meetings/2022-03-14.md

```diff
@@ -0,0 +1,108 @@
+* AC: The first thing is [AS looks up the WebID profile](https://solid.github.io/solid-oidc/#basic-flow) to check the issuer is trusted by this WebID profile. Because I can write a valid ID Token claiming to be anyone. How can an AS know that this particular ID Token is to be trusted.
+* eP: We discussed having a public key that could prove the token was issued by someone trusted.
+* AC: We would have to be careful about token rotation.
+* eP: Yes, with per user key it's probably a less frequent need, but it would indeed put some management burden on the user since they would have to do it.
+* MB: If this scenario is one that we would support, yes the user would need to manage key rotation. 
+* eP: If you want to avoid rotating that, it's a kind of trade-off. There could be a standard period for the user to update their profile. Some services prompt a user to change their password regularly, it seems reasonable.
+* MB: Issuer restrictions would get more complex here. If the issuer is just in charge of managing key pairs to issue tokens, it could even be a piece of software on your machine.
+* eP: Do you refer to self-issued OIDC like flows?
+* MB: Yes, it is de-facto the same kind of thing (or a possible implementation thereof).
+* eP: How to follow up on this issue/conversation? Is anyone interested in proposing initial text of how to address it?
+* MB: It might be interesting to involve people behind WebID-TLS. The main problem there was a usability issue, which was never resolved on the browser level. If that could be resolved server side, maybe WebID-TLS could be a good candidate?
+* eP: Would you like to reach out Matthieu? I'll try to reach out to the self-sovereign identity people.
+* MB: Yes, I'll reach out.
+* AC: In terms of scope I'd like to make sure that we make it clear what is in scope for Solid-OIDC. I think self-issued OIDC is much more in scope. When it comes to WebID-TLS I would see it outside of the scope. I would rather see multiple documents rather than one complicated with forking options.
+* MB: At this stage, I only mentioned WebID-TLS just because it relies on public/private key pair. My intention is to ask: what about managing private keys on the server side? Maybe the two aren't incompatible.
+* AC: One thing to keep in mind. When it comes to existing OIDC infrastructure and ecosystem. The more requirements we put on OP, the harder it is for existing ecosystem to adjust. Currently, the requirements are:
+* ...: 1. WebID scope (fairly typical, adding new scopes to OPs);
+* ...: 2. DPoP which is draft and moving through to the OAuth working group;
+* ...: 3. Verifying client identifiers which is a little bit of a challenge for new providers to support... (Entirely Solid-OIDC and custom).
+* ...: Anything we add akin to that 3rd point and the more custom requirements we add to OPs, the harder it will be for existing ones in the ecosystem to come along with where we're going.
```

Comment:
```suggestion
* ...: 2. DPoP, which is draft and moving through to the OAuth working group;
* ...: 3. Verifying client identifiers, which is a little bit of a challenge for new providers to support... (Entirely Solid-OIDC and custom).
* ...: Anything we add to that 3rd point, and the more custom requirements we add to OPs, the harder it will be for existing ones in the ecosystem to come along to where we're going.
```